library/core/src/time.rs

```diff
@@ -223,6 +223,68 @@ impl Duration {
         }
     }
 
+    /// Creates a new `Duration` from the specified number of whole seconds and
+    /// additional nanoseconds.
+    ///
+    /// If the number of nanoseconds is greater than 1 billion (the number of
+    /// nanoseconds in a second), then it will carry over into the seconds provided.
+    ///
+    /// Returns `None` if the carry from the nanoseconds overflows the seconds
+    /// counter.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(duration_new_checked)]
+    /// use std::time::Duration;
+    ///
+    /// assert_eq!(Duration::new_checked(5, 0), Some(Duration::new(5, 0)));
+    /// assert_eq!(Duration::new_checked(u64::MAX, u32::MAX), None);
+    /// ```
+    #[unstable(feature = "duration_new_checked", issue = "125748")]
+    #[inline]
+    #[must_use]
+    pub const fn new_checked(secs: u64, nanos: u32) -> Option<Duration> {
+        if nanos < NANOS_PER_SEC {
+            // SAFETY: nanos < NANOS_PER_SEC, therefore nanos is within the valid range
+            Some(unsafe { Duration::new_unchecked(secs, nanos) })
```

Comment:
Better construct the duration here inline

Reply:
Done.